### PR TITLE
Feed digest updates

### PR DESCRIPTION
**Pull request by joshm1**
filter digest recos further; lazy load uri summary and attribution details individually instead of loading them in batch

refactor view and introduction of a Context object to encapsulate common dynamic email values ( @2is10 I know you don't like that name and I understand with your reasoning :) may rename in next commit )

### CODE REVIEW

**Comment by joshm1** [comment]
@stkem

**Comment by stkem** [comment]
:+1:

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/email/FeedDigestEmailSender.scala [hunk 1]
```diff
@@ -103,27 +117,32 @@ class FeedDigestEmailSenderImpl @Inject() (
...
+    }
+    digestRecoMailF.flatten
   }
 
-  private def composeAndSendEmail(user: User, digestRecos: Seq[DigestReco], unsubscribeUrl: String): Future[DigestRecoMail] = {
+  private def composeAndSendEmail(user: User, digestRecos: Seq[DigestReco], _unsubscribeUrl: String): Future[DigestRecoMail] = {
     val userId = user.id.get
     val emailData = AllDigestRecos(toUser = user, recos = digestRecos)
-    val htmlBody: LargeString = views.html.email.feedDigest(emailData, unsubscribeUrl).body
+    val ctx = Context(campaign = "emailDigest", unsubscribeUrl = _unsubscribeUrl)
```
Comment (2is10) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/email/FeedDigestEmailSender.scala:143]: I believe adding the underscore was unnecessary. If it was a stylistic choice, I can go along, but personally I’d probably only mangle a name when necessary.

**Comment by 2is10** [comment]
very nice

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/email/FeedDigestEmailSender.scala (hunk 1 above)
Comment (joshm1) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/email/FeedDigestEmailSender.scala:143]: I'm not a fan of this either. Only did b/c for some reason that I can't reproduce right now I thought there was a conflict I tried using named variables and variables of the same name. But I just renamed and it compiled, so nevermind that

**Pull request merged**